crates/provider/src/provider/web3_signer.rs

```diff
@@ -0,0 +1,176 @@
+use crate::fillers::{FillerControlFlow, TxFiller};
+use alloy_eips::Decodable2718;
+use alloy_network::{Ethereum, Network, TransactionBuilder};
+use alloy_primitives::{Address, Bytes};
+use alloy_transport::{TransportErrorKind, TransportResult};
+
+use super::{DynProvider, Provider, SendableTx};
+
+/// A remote signer that leverages the underlying provider to sign transactions using
+/// `"eth_signTransaction"` requests.
+///
+/// For more information, please see [Web3Signer](https://docs.web3signer.consensys.io/).
+///
+/// [`Web3Signer`] also implements [`TxFiller`] to allow it to be used as a filler in the
+/// [`ProviderBuilder`].
+///
+/// Note:
+///
+/// `"eth_signTransaction"` is not supported by regular nodes.
+///
+/// [`ProviderBuilder`]: crate::ProviderBuilder
+#[derive(Debug, Clone)]
+pub struct Web3Signer<N: Network = Ethereum> {
+    /// The provider used to make `"eth_signTransaction"` requests.
+    provider: DynProvider<N>,
```

Comment:
this can now also be used as regular provider, but we don't expose the underlying provider so all this type can do sign stuff and not send stuff